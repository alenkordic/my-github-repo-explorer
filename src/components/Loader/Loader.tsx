import React from "react";
import Box from "@mui/material/Box";
import { Typography, LinearProgress } from "@mui/material";

type LoaderProps = {
  text?: string
};

const Loader = ({ text = "Loading..." }: LoaderProps): JSX.Element => {
  return (
    <Box sx={{ width: 400, margin: "auto 0" }} mt={50}>
      <LinearProgress color="primary" />

      <Typography variant="body2" textAlign="center" mt={2}>
        {text}
      </Typography>
    </Box>
  );
};

export default Loader;
