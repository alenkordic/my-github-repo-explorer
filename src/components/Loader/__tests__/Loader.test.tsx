import React from "react";
import Loader from "../Loader";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

test("Loader renders well", () => {
  render(<Loader text="Loader testing" />);
  expect(screen.getByText("Loader testing")).toBeTruthy();
});

//Snapshots
it("Loader checked", () => {
  const tree = renderer.create(<Loader />).toJSON();
  expect(tree).toMatchSnapshot();
});
