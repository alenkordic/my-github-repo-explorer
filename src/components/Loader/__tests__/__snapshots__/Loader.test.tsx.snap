// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader checked 1`] = `
<div
  className="MuiBox-root css-tczqft"
>
  <span
    className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate css-eglki6-MuiLinearProgress-root"
    role="progressbar"
  >
    <span
      className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate css-5ir5xx-MuiLinearProgress-bar1"
      style={Object {}}
    />
    <span
      className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar2Indeterminate css-1r8wrcl-MuiLinearProgress-bar2"
      style={Object {}}
    />
  </span>
  <p
    className="MuiTypography-root MuiTypography-body2 css-1sgvizo-MuiTypography-root"
  >
    Loading...
  </p>
</div>
`;
